test/evergreen/evg_cfg.py

```diff
@@ -0,0 +1,327 @@
+#! /usr/bin/env python3
+
+"""
+This program provides an CLI interface to check and generate Evergreen configuration.
+"""
+
+import os
+import sys
+import re
+import subprocess
+
+try:
+    import docopt
+except Exception as e:
+    modules = "docopt"
+    print("ERROR: %s" % e)
+    print("Use pip to install the required library:")
+    print("  pip install %s" % modules)
+    sys.exit(0)
+
+TEST_TYPES = ('make_check', 'csuite')
+EVG_CFG_FILE = "test/evergreen.yml"
+CSUITE_TEST_DIR = "test/csuite"
+MAKE_CHECK_TEST_TMPLT = "test/evergreen/make_check_test_evg_task.template"
+CSUITE_TEST_TMPLT = "test/evergreen/csuite_test_evg_task.template"
+MAKE_CHECK_TEST_SEARCH_STR = "  # End of normal make check test tasks"
+CSUITE_TEST_SEARCH_STR = "  # End of csuite test tasks"
+WIREDTIGER_REPO = "git@github.com:wiredtiger/wiredtiger.git"
+
+# This list of sub directories will be skipped from checking.
+# They are not expected to trigger any 'make check' testing.
+make_check_subdir_skips = [
+    "api/leveldb",  # no need to test
+    "test/csuite",  # csuite has its own set of Evergreen tasks, skip the checking here
```

Comment:
Should this be `test/`, since the evergreen file is at that level, not at `test/csuite`?

Reply:
`test/csuite` will be queried out in the new approach as its Makefile.am contains `TESTS =`. The reason the `test/csuite` (make check) directory is put into skip list is the csuite tests under it had been extended to separate Evergreen tasks, and we don't want to duplicate the csuite tests again with a `csuite-test` Evergreen task. The change is aimed to advise updates needed for `test/evergreen.yml` - the main Evergreen configuration file for WiredTiger (Evergreen) project - when new make check test applicable directory is introduced.